### PR TITLE
[Bugfix] Empty value in Jelix checkboxes 

### PR DESCRIPTION
**Pull request by rldhont**
* Partially revert changes from *[Bugfix] Empty value in Jelix menulist* https://github.com/3liz/lizmap-web-client/pull/2824
* Add end2end tests for ValueRelation allowMulti widgets
* [Edition] AllowMulti: remove unnecessary spaces between checkboxes

### CODE REVIEW

**Comment by Gustry** [comment]
I think we could remove some lines in the `form_type_relational_value.md` ? @nboisteault

File: lizmap/modules/lizmap/lib/Form/QgisForm.php [hunk 1]
```diff
@@ -1245,15 +1245,22 @@ private function fillControlFromValueRelationLayer($fieldName, $formControl)
...
         $formControl->ctrl->emptyItemLabel = '';
-        if ($formControl->valueRelationData['allowMulti'] && $formControl->ctrl->required) {
-            $formControl->ctrl->emptyItemLabel = null;
-        }
+
+        // Create Datasource
         $dataSource = new QgisFormValueRelationDynamicDatasource($formControl->ref);
 
+        // In lib/jelix/plugins/formwidget/checkboxes_html/checkboxes_html.formwidget.php
+        // control emptyItemLabel or required is not tacking into account
```
Comment (nboisteault) [inline lizmap/modules/lizmap/lib/Form/QgisForm.php:1255]: @rldhont i don't understand this sentence, do you mean:
```suggestion
        // control emptyItemLabel where required is not taken into account
```

Comment (Gustry) [reply lizmap/modules/lizmap/lib/Form/QgisForm.php:1255]: Lets say yes, fix typo layer if needed please @rldhont

Comment (rldhont) [reply lizmap/modules/lizmap/lib/Form/QgisForm.php]: It was not `emptyItemLabel where required` but more `emptyItemLabel not null or required positif`

**Pull request merged**